app/internal_packages/undo-redo/lib/undo-redo-toast.tsx

```diff
@@ -11,6 +12,29 @@ function isUndoSend(block) {
   );
 }
 
+async function sendMessageNow(block) {
+  if (isUndoSend(block)) {
+      const message = await DatabaseStore.find<Message>(Message, (block.tasks[0] as SyncbackMetadataTask).modelId),
+        newExpiry = Math.floor(Date.now() / 1000);
+
+      Actions.queueTask(
+        SyncbackMetadataTask.forSaving({
+          model: message,
+          pluginId: PLUGIN_ID,
+          value: {
+            expiration: newExpiry,
+          },
+        })
+      );
+      
+      block.tasks[0].value.expiration = newExpiry;
```

Comment:
This looks good, cool that updating the metadata actually causes it to send immediately. I'll try this a couple times once we merge just to stress test it a bit, but it seems safe to me.